src/Stratis.Bitcoin.Features.IndexStore/IndexStoreFeature.cs

```diff
@@ -28,10 +31,24 @@ public override BlockStoreBehavior BlockStoreBehaviorFactory()
         }
     }
 
-    public static class IndexStoreBuilderExtension
+    public static class FullNodeBuilderIndexStoreExtension
     {
         public static IFullNodeBuilder UseIndexStore(this IFullNodeBuilder fullNodeBuilder, Action<IndexSettings> setup = null)
         {
+            try
```

Comment:
>If I'm not mistaken services will not get resolved until they are called.

Correct. Problem is that they get called for the feature creation that should actually perform the dependency check after it is created. So in case we use mempool without consensus that's what happening: 

Node gets builded without consensus, then node.Start() => fullNodeFeatureExecutor.Start(); which calls feature constructor that need MempoolBehavior feature. And MempoolBehavior needs MempoolOrphans which needs Mempoolvalidator which needs PowConsensusValidator which is a part of consensus feature aaand it fails.

Reply:
>This is probably not the right way to check dependencies, if doing it inside the feature is not good its should probably be done some where inside he builder itself.

Ok, I'll remake the PR. 
Also I guess I need to remove old dependency checking (since it gets called too late) and move it's usages to a new dependency checking